include/mbedtls/x509_csr.h

```diff
@@ -77,6 +77,19 @@ typedef struct mbedtls_x509write_csr {
 }
 mbedtls_x509write_csr;
 
+typedef struct mbedtls_x509_san_node {
+    int type;                       /**< Subject Alternative Name types */
+    char *name;                     /**< Value, following the syntax allowed bythe type */
+    size_t len;                     /**< Length of the provided value */
+}
+mbedtls_x509_san_node;
+
+typedef struct mbedtls_x509_san_list {
```

Comment:
I think that the `mbedtls_asn1_sequence` might be used here instead. It has a buffer and a pointer to a next element, and inside you have an `mbedtls_asn1_buf` with a tag, length and a pointer to data, just as the `mbedtls_x509_san_node`.

Reply:
Ignore my previous comment. When we'll expand to use other types, `mbedtls_x509_san_node` will need support for other types as well, and `*name` will change its type to `void*` instead. reusing `mbedtls_asn1_sequence` for this purpose would be a hack.